pkg/controller/acmechallenges/sync.go

```diff
@@ -396,7 +396,17 @@ func (c *controller) acceptChallenge(ctx context.Context, cl acmecl.Interface, c
 	// blocking the challenge's processing. Here, we defensively add a timeout for this exchange
 	// with the ACME server and a "context deadline reached" error will be returned by WaitAuthorization
 	// in the err variable.
-	ctxTimeout, cancelAuthorization := context.WithTimeout(ctx, authorizationTimeout)
+	var ato time.Duration
+	if atoOriginal != "" {
+		ato, err = time.ParseDuration(atoOriginal)
+		if err != nil {
+			log.Error(err, "invalid authorizationTimeout in (Cluster)Issuer")
+			return handleError(ch, err)
+		}
+	} else {
+		ato = authorizationDefaultTimeout
+	}
+	ctxTimeout, cancelAuthorization := context.WithTimeout(ctx, ato)
```

Comment:
Can you describe a bit more why your use case takes more than 10s to complete this step? 10s seems like a very long time already - what does the acme server do here/what does the spec say a reasonable timeout value is? I'd lean toward increasing this timeout globally if we deem acceptChallenge may take a while (or look at making this step asynchronous rather than allowing the controller to block for longer).

Reply:
I searched quite a time but did not find a clear reason. I guess its a combination of
- let´s encrypt staging environment that takes a time to release a certificate after approval via challenge
- my environment being not very powerful  and heavily loaded during CI (spinning up some VM´s and a Kubernetes cluster from scratch)
- my network seems slow sometimes, maybe busy with video-streaming

It does not always take longer than 10s but if I run my CI I don´t like it to fail occasionally just for that reason. Having the patch active since a week (timeout set to 2min) I did not see this issue anymore.

Reply:
My concern is that making this configurable allows someone to enter some invalid details/ACME server here which is intentionally slow to respond, causing the 'acmeorders' controllers to block for up to the user configurable limit at a time. This is a DoS vector if it's end-user configurable.

Accepting an authorisation happens after self checks are passing, so your issue here implies your self check is passing but later requests from the acme server fail... probably some kind of propagation delay, or maybe you have reconfigured your self checks to bypass an external load balancer or something?

Reply:
Hmm, this someone needs to have the rights to create an Issuer or clusterIssuer in the Kubernetes cluster. Then he can probably do other things even worse. 
Anyhow, an asynchronous way of waiting for the ACME server is in any case better, preventing the DoS vector no matter how risky it is and improving throughput. I´m not sure if my go skills are enough to do it, anyhow I would give it a try...

Having a DSLite without public IPv4 there is some delta between internal and external view on the ingress that I cannot avoid. It may lead to propagation delay as you said, on top there is some slowness as described above.